sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/object.scala

```diff
@@ -719,3 +720,99 @@ case class CoGroup(
   override protected def withNewChildrenInternal(
       newLeft: LogicalPlan, newRight: LogicalPlan): CoGroup = copy(left = newLeft, right = newRight)
 }
+
+object JoinWith {
+  /**
+   * find the trivially true predicates and automatically resolves them to both sides.
+   */
+  private[sql] def resolveSelfJoinCondition(resolver: Resolver, plan: Join): Join = {
+    val cond = plan.condition.map {
```

Comment:
If you need a reason why this should be pushed into the analyzer then this would be it :)